lib/tools/src/cover.erl

```diff
@@ -2834,6 +2831,222 @@ do_analyse_to_file1(Module, OutFile, ErlFile, HTML) ->
+    %% the summary file name.
+    SrcPaths = lists:usort([filename:dirname(File) || {_, File} <- Modules]),
+    Title =
+        case
+            [filelib:wildcard(Path ++ "/*.{app,app.src}") || Path <- SrcPaths]
+        of
+            [AppFile] ->
+                try
+                    {ok, [{application, AppName, _}]} = file:consult(AppFile),
+                    "Coverage summary for " ++ atom_to_list(AppName)
+                catch
+                    _:_ ->
+                        SummaryFile
+                end;
+            _L ->
+                SummaryFile
+        end,
+
+    {FileOk, FileErr} =
+        case file:open(SummaryFile, [write, raw, delayed_write]) of
+            {ok, OutFd} ->
+                %% Write header
+                if HTML ->
+                       Header = create_header(Title, Enc, true),
+                       HeaderBin =
+                           unicode:characters_to_binary(Header, Enc, Enc),
+                       ok = file:write(OutFd, HeaderBin);
+                   true ->
+                       %% Omit plaintext header if app name wasn't found
+                       if Title =/= SummaryFile ->
+                              Header = [Title, "\n\n"],
+                              HeaderBin =
+                                  unicode:characters_to_binary(Header,Enc,Enc),
+                              ok = file:write(OutFd, HeaderBin);
+                          true ->
+                              ok
+                       end
+                end,
+
+                %% Write some initial information to the output file
+                ok = file:write(OutFd, outfileinfo(undefined, Enc, HTML, true)),
+
+                %% Column headers for plaintext summary
+                summary_create_header(OutFd, HTML),
+
+                %% Print a line for each module and sum total coverage
+                PrintSumFun =
+                    fun({Module, {Cov, NotCov}}, {TotCovAcc, TotNotCovAcc}) ->
+                            {summary_print_line(OutFd, HTML,
+                                                Module, Cov, NotCov),
+                             {Cov + TotCovAcc, NotCov + TotNotCovAcc}}
+                    end,
+
+                {_, {TotCov, TotNotCov}} =
+                    lists:mapfoldr(PrintSumFun, {0,0}, CovRes),
+
+                %% Print totals
+                ok = file:write(OutFd, create_totals(HTML, TotCov, TotNotCov)),
+
+                if HTML ->
+                       ok = file:write(OutFd, summary_close_html());
+                   true ->
+                       ok
+                end,
+
+                ok = file:close(OutFd),
+
+                {[{ok, SummaryFile}], []};
+            {error, Reason} ->
+                {[], [{error, {file, SummaryFile, Reason}}]}
+        end,
+
+    FileOk ++ CovErr ++ FileErr.
+
+
+-spec summaryfilename(SummaryFilename :: list(), HTML :: boolean()) ->
+    SummaryFilenameRes :: list().
+summaryfilename(Opts) ->
```

Comment:
Perhaps name this function `summary_filename`?

Reply:
I prefer this style, I just followed how outfilename was named.

I'll change it.